### ISSUE TITLE
Extending to multi-colored nonograms

### ISSUE THREAD

**Issue by helq** [attilaszia/nonogram#6]
According to Batenburg and Kosters (2009), the algorithm allows for more than two colors (black and white). I want to modify the implementation to allow for multicolored nonograms.

But looking at the code, it seems like this is NOT easy to implement. Correct me if I'm wrong, but the entirety of the implementation assumes only two possible colors, black and white, and thus adding a new color would require a complete refactoring of the entire code base first.

How long do you think it would take to refactor the code to allow for more than two colors? (Ignoring the extra code to load a new file format and the png generation. Those are easy.)

Thanks!

**Comment by attilaszia** [comment]
Hey, if this is still relevant, I'm doing some maintenance work on the codebase currently

**Comment by helq** [comment]
Nah, don't worry. I no longer need it. Thanks for checking though
